pkg/config/config.go

```diff
@@ -96,63 +108,110 @@ func LoadConfig(lokocfgPath, lokocfgVarsPath string) (*HCL, hcl.Diagnostics) {
 		}
 	}
 
-	hclParser := hclparse.NewParser()
-
-	var hclFiles []*hcl.File
-	for _, f := range lokocfgPaths {
-		hclFile, diags := hclParser.ParseHCLFile(f)
-		if len(diags) > 0 {
-			return nil, diags
+	for _, path := range lokocfgPaths {
+		data, err := loadHCLFile(path)
+		if err != nil {
+			return nil, hcl.Diagnostics{
+				&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  err.Error(),
+				},
+			}
 		}
-		hclFiles = append(hclFiles, hclFile)
+
+		files[path] = data
 	}
 
-	configBody := hcl.MergeFiles(hclFiles)
+	return files, nil
+}
 
-	exists, err := util.PathExists(lokocfgVarsPath)
+// loadHCLfile loads the hcl file provided by the path and returns
+// the contents of the file as a slice of bytes.
+func loadHCLFile(path string) ([]byte, error) {
+	data, err := ioutil.ReadFile(filepath.Clean(path))
 	if err != nil {
-		return nil, hcl.Diagnostics{
-			&hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  fmt.Sprintf("could not stat %q: %v", lokocfgVarsPath, err),
-			},
-		}
-	}
-	var userVals map[string]cty.Value
-	var diags hcl.Diagnostics
-	if exists {
-		userVals, diags = LoadValuesFile(lokocfgVarsPath)
-		if len(diags) > 0 {
-			return nil, diags
-		}
+		return nil, err
 	}
 
-	var clusterConfig ClusterConfig
-	diags = gohcl.DecodeBody(configBody, nil, &clusterConfig)
-	if len(diags) > 0 {
-		return nil, diags
+	return data, nil
+}
+
+// getHCLFiles takes a map of file name and its contents and returns
+// a slice of HCL File instance.
+func getHCLFiles(files map[string][]byte) ([]*hcl.File, hcl.Diagnostics) {
+	var diagnostics hcl.Diagnostics
+
+	hclFiles := []*hcl.File{}
+
+	hclParser := hclparse.NewParser()
+
+	for path, content := range files {
+		f, diags := hclParser.ParseHCL(content, path)
+		diagnostics = append(diagnostics, diags...)
+		hclFiles = append(hclFiles, f)
 	}
 
+	return hclFiles, diagnostics
+}
+
+// parseVariables parses the variables used in the HCL configuration.
+func parseVariables(userVals map[string]cty.Value, cfg ClusterConfig) (map[string]cty.Value, hcl.Diagnostics) {
```

Comment:
I can't figure out exactly what this function does, but I get the impression "parse" isn't the right description. Could we name this function better and be more explicit regarding its purpose?